### ISSUE TITLE
Plans for a php7 version?

### ISSUE THREAD

**Issue by trq** [container-interop/container-interop#38]
I am working on a project using php7, and am currently using your interfaces without type hints. Would you welcome a pr with scaler type hints in place?

**Comment by mnapoli** [comment]
Hi! It would introduce some issues: we would need to create/maintain a 2.0 branch + releases and risk splitting one standard into 2 standards. Also, there are no containers that I know of that are PHP 7 only (so no immediate benefits since the standard would be implemented or used by no-one).

I think we should wait a little more with this.

**Comment by shochdoerfer** [comment]
In addition to that we probably should wait and see how the FIG wants to tackle the problem as the problem exists for the other PSR`s as well.

**Comment by trq** [comment]
Part of my project actually builds a new container so I have a valid use case. However, I completely understand your position. Will wait and see.

**Comment by njasm** [comment]
I agree with @mnapoli completely, it make sense what @trq says, but probably not right now..

**Comment by Ocramius** [comment]
IMO this issue should be reconsidered after PHP 5.6 EOL. At that point, it would be safe to create a 2.0 branch, as the 1.0 branch would be stable enough to not require major changes.

**Comment by Ocramius** [comment]
@mnapoli can you eventually create a 2.0.0 milestone, so that we can assign this issue to it?

**Comment by mnapoli** [comment]
Done!

**Comment by Ocramius** [comment]
Thanks

**Comment by Sam-Burns** [comment]
With PHP 7 now being stable, would there be any interest in reopening this discussion?  It would be great to see a ```2.0``` branch with service IDs typehinted as ```string```s, and a return typehint on the ```has()``` method.

We're still a year and a half away from PHP5 being completely dead, and it seems like a long time to wait for those of us using PHP7 in production now.  I'm sure the two branches would be pretty similar, other than the typehints and the ```composer.json``` file.  I'd be more than happy to write a pull request, if that would help??

**Comment by mnapoli** [comment]
I'm a huge fan and proponent of PHP 7 but I honestly see very little value in this:

- I don't see implementations (i.e. containers) that are PHP 7 only (with significant usage)
- I don't see consumers (e.g. frameworks) that are PHP 7 only (with significant usage)
- it would split the standard in 2 (container interop v1 and v2)
- the actual gain for developers is insignificant: I don't believe those 3 new scalar typehints will change anything in terms of behavior in consumers/implementations
- typehints are currently covered by the phpdoc: if you don't comply with the phpdoc, you are not container-interop compatible

I'm definitely -1 until the situation changes and PHP 7 is widespread (or we have a separate reason to release a v2 and we think it's fine to make it PHP 7 only).

**Comment by keithmifsud** [comment]
I agree with being to declare/use strict types on the interfaces. I'm currently trying to implement all of the container-interop interfaces on a new DI package I'm developing and this is breaking my own code quality. As it stands we cannot enforce the arguments to be (for example) a string. This flaw gets extended by other interfaces and so on throughout the package and then the framework.

It does not make sense not to use PHP7 now considering there's PHP7.1 coming soon. I have required PHP>=7 in composer schema of every new project from the start of the year. I am sure a lot of other devs are doing the same.

**Comment by mnapoli** [comment]
> It does not make sense not to use PHP7 now considering there's PHP7.1 coming soon.

It does make sense: backward compatibility.

**Comment by keithmifsud** [comment]
Thanks @mnapoli, apologies. I didn't realise that we already have a stable release.

**Comment by mnapoli** [comment]
@keithmifsud no worries, my apologies if it sounded a bit sharp I'm currently going through lots of tickets so I went to the point ;)

**Comment by keithmifsud** [comment]
@mnapoli No worries here either. My comment was also a bit abrupt :) Am I right to assume that we will try to not release v2.0 before FIG have done their research?
